exporter/redis.go

```diff
@@ -189,9 +196,39 @@ func (e *Exporter) initGauges() {
 	}, []string{"addr", "alias"})
 }
 
+// splitKeyArgs splits a command-line supplied argument into a slice of dbKeyPairs.
+func parseKeyArg(keysArgString string) (keys []dbKeyPair, err error) {
+	if keysArgString == "" {
+		return keys, err
+	}
+	for _, k := range strings.Split(keysArgString, ",") {
+		db := "0"
+		key := ""
+		frags := strings.Split(k, "=")
+		switch len(frags) {
+		case 1:
+			db = "0"
+			key, err = url.QueryUnescape(strings.TrimSpace(frags[0]))
+		case 2:
+			db = strings.Replace(strings.TrimSpace(frags[0]), "db", "", -1)
+			key, err = url.QueryUnescape(strings.TrimSpace(frags[1]))
+		default:
+			err = fmt.Errorf("more than one '=' in key specification in key list argument in: %s", k)
+			return keys, err
+		}
+		if err != nil {
+			err = fmt.Errorf("couldn't parse db/key string: %s", k)
+			return keys, err
+		}
+
+		keys = append(keys, dbKeyPair{db, key})
+	}
+	return keys, err
+}
+
 // NewRedisExporter returns a new exporter of Redis metrics.
 // note to self: next time we add an argument, instead add a RedisExporter struct
```

Comment:
Not sure exactly what you mean by this, glad to do whatever with some more direction.